### PR TITLE
init ui and fetch music genres data

### PR DESCRIPTION
**Pull request by thaohq-0814**
init ui and fetch music genres data
Link: https://edu-redmine.framgia.vn/issues/5472
image

### CODE REVIEW

File: app/src/main/java/com/framgia/music_50/data/source/local/TrackLocalDataSource.java [hunk 1]
```diff
@@ -11,4 +16,14 @@ public static TrackLocalDataSource getInstance() {
         }
         return sInstance;
     }
+
+    @Override
+    public void getMusicGenres(OnFetchDataListener<Genre> listener) {
+        List<Genre> genreList = new ArrayList<>();
+        genreList.add(new Genre("Alternative Rock", R.drawable.alternative_rock_genre));
```
Comment (phuongohn-1626) [inline app/src/main/java/com/framgia/music_50/data/source/local/TrackLocalDataSource.java]: hard code and below

File: app/src/main/res/layout/item_layout_genre.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        android:layout_height="0dp"
+        android:layout_marginBottom="@dimen/dp_24"
+        android:layout_marginTop="@dimen/dp_24"
+        app:cardElevation="0dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintDimensionRatio="@string/genre_item_dimension_ratio"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintWidth_percent=".6"
```
Comment (phuongohn-1626) [inline app/src/main/res/layout/item_layout_genre.xml]: 0.6

**Review by buivanhieu7112 — APPROVED** [review]

**Comment by thaohq-0814** [comment]
ready

File: app/src/main/java/com/framgia/music_50/screen/home/adapter/GenreAdapter.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import android.widget.TextView;
+import com.bumptech.glide.Glide;
+import com.framgia.music_50.R;
+import com.framgia.music_50.data.model.Genre;
+import com.framgia.music_50.utils.OnItemRecyclerViewClickListener;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GenreAdapter extends RecyclerView.Adapter<GenreAdapter.ItemMusicGenreViewHolder> {
+    private List<Genre> mGenreList;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_50/screen/home/adapter/GenreAdapter.java]:
```suggestion
    private List<Genre> mGenres;
```

File: app/src/main/java/com/framgia/music_50/screen/home/adapter/GenreAdapter.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    private OnItemRecyclerViewClickListener<Object> mOnItemRecyclerViewClickListener;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_50/screen/home/adapter/GenreAdapter.java]:
```suggestion
    private OnItemRecyclerViewClickListener<Object> mOnItemClickListener;
```

File: app/src/main/java/com/framgia/music_50/screen/home/adapter/GenreAdapter.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    public int getItemCount() {
+        return mGenreList != null ? mGenreList.size() : 0;
+    }
+
+    public void setOnItemRecyclerViewClickListener(
+            OnItemRecyclerViewClickListener<Object> onItemRecyclerViewClickListener) {
+        mOnItemRecyclerViewClickListener = onItemRecyclerViewClickListener;
+    }
+
+    public void updateData(List<Genre> list) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_50/screen/home/adapter/GenreAdapter.java]:
```suggestion
    public void updateData(List<Genre> genres) {
```

File: app/src/main/java/com/framgia/music_50/screen/home/adapter/GenreAdapter.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        mGenreList.clear();
+        mGenreList.addAll(list);
+        notifyDataSetChanged();
+    }
+
+    static class ItemMusicGenreViewHolder extends RecyclerView.ViewHolder
+            implements View.OnClickListener {
+        private ImageView mGenreImageView;
+        private TextView mGenreTitleTextView;
+        private List<Genre> mGenreList;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_50/screen/home/adapter/GenreAdapter.java]:
```suggestion
        private List<Genre> mGenres;
```

**Comment by thaohq-0814** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**